scripts/cbmc_json_parser.py

```diff
@@ -83,22 +88,47 @@ def main(argv):
     if not output_style:
         usage_error(f"Invalid output format '{argv[2]}'.")
 
+    # Check the mode of the output
+    mode = os.fstat(sys.stdin.fileno()).st_mode
     extra_ptr_check = False
+
+    # The cbmc output can be either file mode (default) or stream mode (--read-cbmc-from-stream)
+    cbmc_output_mode = argv[1].split(".")[-1]
+    if not ((cbmc_output_mode == "cbmc_output") or (cbmc_output_mode == GlobalMessages.READ_FROM_STREAM)):
+        usage_error(f"CBMC output file or mode invalid\n")
+
     if len(argv) == 4:
         if argv[3] == "--extra-ptr-check":
             extra_ptr_check = True
         else:
             usage_error(f"Unexpected argument '{argv[3]}'.")
 
-    # parse the input json file
-    with open(argv[1]) as f:
-        sample_json_file_parsing = f.read()
+    if argv[1] == GlobalMessages.READ_FROM_STREAM:
+        # Print each streaming line live from cbmc without writing to file
+        # temporary output, will change to be parsed like the regular json objects
+        # that are written to a file
 
-    # the main function should take a json file as input
-    return_code = transform_cbmc_output(sample_json_file_parsing,
-                                        output_style, extra_ptr_check)
-    sys.exit(return_code)
+        # Check if the stdin input comes from the pipe
+        if S_ISFIFO(mode):
+            try:
+                for line in sys.stdin:
+                    print(line)
+            except BaseException:
+                usage_error(f"Unable to complete parsing CBMC Output\n")
+        else:
+            usage_error(f"CBMC Output not piped to post-processing\n")
+        return_code = 0
```

Comment:
I would use an error code here.

Reply:
I've added a custom exit string and an error code of 1 with - 

```
print(f"Unable to complete parsing CBMC Output\n")
sys.exit(1)
```

Would this work?

Reply:
Sorry, my comment here was not very clear.

If I'm not mistaken, this will print all lines from CBMC's output and return 0. If we change this to 1, do the regression tests still pass?

Reply:
I should have clarified more. The return code is 1 if the stream of output is interrupted , either by a broken line or broken pipe or user interrupt (Ex - keyboard interrupt). If it completes printing all the lines from CBMC, then it returns 0. 

I hope this is an acceptable usage of the error code?

Reply:
A return code with value 0 basically means that verification is always successful.